cashweb-registry/src/http/server.rs

```diff
@@ -218,3 +238,50 @@ async fn handle_get_metadata_range(
             .collect(),
     }))
 }
+
+async fn handle_get_messages(
+    Path(topic): Path<String>,
+    Query(params): Query<MessagesQuery>,
+    Extension(server): Extension<RegistryServer>,
+) -> Result<Protobuf<cashweb_payload::proto::SignedPayloadSet>, HttpRegistryError> {
+    let signed_payloads = server
+        .registry
+        .get_messages(topic.as_str(), params.from, params.to)?
+        .iter()
+        .map(|signed_payload| signed_payload.to_proto())
+        .collect();
+
+    let payload_page = SignedPayloadSet {
+        items: signed_payloads,
+    };
+
+    Ok(Protobuf(payload_page))
```

Comment:
The API here should probably get hashes instead of full messages. It also needs better paging support.